draft-ietf-quic-transport.md

```diff
@@ -1474,31 +1506,33 @@ validate a client address, it provides the client with a token.  As long as the
 token cannot be easily guessed (see {{token-integrity}}), if the client is able
 to return that token, it proves to the server that it received the token.
 
-During the processing of the cryptographic handshake messages from a client, TLS
-will request that QUIC make a decision about whether to proceed based on the
-information it has.  TLS will provide QUIC with any token that was provided by
-the client.  For an initial packet, QUIC can decide to abort the connection,
-allow it to proceed, or request address validation.
-
-If QUIC decides to request address validation, it provides the cryptographic
-handshake with a token.  The contents of this token are consumed by the server
-that generates the token, so there is no need for a single well-defined format.
-A token could include information about the claimed client address (IP and
-port), a timestamp, and any other supplementary information the server will need
-to validate the token in the future.
-
-The cryptographic handshake is responsible for enacting validation by sending
-the address validation token to the client.  A legitimate client will include a
-copy of the token when it attempts to continue the handshake.  The cryptographic
-handshake extracts the token then asks QUIC a second time whether the token is
-acceptable.  In response, QUIC can either abort the connection or permit it to
-proceed.
+If QUIC decides to request address validation, it encodes the token in a Retry
+packet.  The contents of this token are consumed by the server that generates
+the token, so there is no need for a single well-defined format.  A token could
+include information about the claimed client address (IP and port), a
+timestamp, and any other supplementary information the server will need to
+validate the token in the future. Servers SHOULD construct tokens in such
+a way that they cannot be forged, for instance by AEAD-encrypting them
+with a key known to the server.
```

Comment:
and rotate the key so that tokens cannot be collected?

Reply:
Doesn't the expiration time address this.